llvm/lib/IR/StructuralHash.cpp

```diff
@@ -121,14 +148,17 @@ class StructuralHashImpl {
       // This random value acts as a block header, as otherwise the partition of
       // opcodes into BBs wouldn't affect the hash, only the order of the
       // opcodes
-      hash(45798);
+      Hashes.emplace_back(45798);
```

Comment:
Can we hash a string (maybe "BasicBlock Start"?) rather than an arbitrary number to make it more obvious what we are doing? Same for `23456` below.

Reply:
The functions end up getting sorted by hash, so can cause test failures if you aren't careful (and I remember running into issues like 32-bit vs 64-bit x86 differences there). I'd prefer to keep explicit constants, but refactoring them to named constants would make a lot of sense to me.

Reply:
I have now hashed the name. The same approach is applied to Function and Global. Accordingly, I also updated the affected tests.

Reply:
`stable_hash_name` is not `constexpr`, so might impose a reasonable runtime performance penalty if the optimizer isn't able to optimize everything and constant fold it all (which seems like it could be unlikely).

I would still prefer moving the values to named constants, unless there's some other reason I'm missing to do it this way.

Reply:
I updated the code to compute the values once during the construction of the StructuralHashImpl instance. I believe this is a reasonable trade-off for improved readability while minimizing performance penalties.

Reply:
Is there a reason we can't mark these `constexpr` and use the existing sentinel values (or choose arbitrary new ones) to completely avoid having to call `stable_hash_name`? I believe the variable name along should provide the same level of readability.